src/ORM/Association/HasMany.php

```diff
@@ -134,6 +181,63 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     * Skips deleting records present in $remainingEntities
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $remainingEntities Entities that should not be deleted
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities)
+    {
+        $primaryKey = (array)$target->primaryKey();
+        $mustBeDependent = (!$this->_foreignKeyAcceptsNull($target, $properties) || $this->dependent());
+        $conditions = [
+            'AND' => [
+                'NOT' => [
+                    'AND' => array_map(
```

Comment:
@lorenzo I just noticed I was using the wrong operator all the time, you were right in your first statement. I just fixed, cause it should be OR instead of AND.

The SQL it produces is:
https://gist.github.com/mylux/f6dd567b47525720f40a